src/test/java/ca/uhn/fhir/jpa/starter/interceptor/CustomAuthorizationInterceptorTest.java

```diff
@@ -0,0 +1,708 @@
+import org.mockito.MockedStatic;
+import org.mockito.MockitoAnnotations;
+import org.mockito.Spy;
+import org.mockito.stubbing.Answer;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.auth0.jwt.exceptions.TokenExpiredException;
+import com.auth0.jwt.interfaces.DecodedJWT;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import ca.uhn.fhir.context.FhirContext;
+import ca.uhn.fhir.interceptor.api.HookParams;
+import ca.uhn.fhir.interceptor.api.IInterceptorBroadcaster;
+import ca.uhn.fhir.interceptor.api.Pointcut;
+import ca.uhn.fhir.jpa.starter.AppProperties;
+import ca.uhn.fhir.jpa.starter.AppProperties.Apikey;
+import ca.uhn.fhir.jpa.starter.AppProperties.Oauth;
+import ca.uhn.fhir.jpa.starter.util.ApiKeyHelper;
+import ca.uhn.fhir.jpa.starter.util.OAuth2Helper;
+import ca.uhn.fhir.rest.annotation.ConditionalUrlParam;
+import ca.uhn.fhir.rest.annotation.Delete;
+import ca.uhn.fhir.rest.annotation.IdParam;
+import ca.uhn.fhir.rest.annotation.OptionalParam;
+import ca.uhn.fhir.rest.annotation.Read;
+import ca.uhn.fhir.rest.annotation.Search;
+import ca.uhn.fhir.rest.api.EncodingEnum;
+import ca.uhn.fhir.rest.api.MethodOutcome;
+import ca.uhn.fhir.rest.api.server.RequestDetails;
+import ca.uhn.fhir.rest.api.server.storage.TransactionDetails;
+import ca.uhn.fhir.rest.param.TokenAndListParam;
+import ca.uhn.fhir.rest.server.IResourceProvider;
+import ca.uhn.fhir.rest.server.RestfulServer;
+import ca.uhn.fhir.rest.server.exceptions.ResourceNotFoundException;
+import ca.uhn.fhir.rest.server.interceptor.auth.IAuthRule;
+import ca.uhn.fhir.rest.server.interceptor.auth.RuleBuilder;
+import ca.uhn.fhir.rest.server.servlet.ServletRequestDetails;
+import ca.uhn.fhir.test.utilities.JettyUtil;
+import ca.uhn.fhir.util.TestUtil;
+
+class CustomAuthorizationInterceptorTest {
+
+	@Spy
+	AppProperties mockConfig;
+
+	@Spy
+	Oauth mockOAuth;
+
+	@Spy
+	Apikey mockApikey;
+
+	@Mock
+	RequestDetails mockRequestDetails;
+
+	@InjectMocks
+	CustomAuthorizationInterceptor ourInterceptor;
+
+	private static Logger logger = LoggerFactory.getLogger(CustomAuthorizationInterceptorTest.class);
+	private static RestfulServer ourServlet;
+	private static boolean ourHitMethod;
+	private static List<Resource> ourReturn;
+	private static int ourPort;
+	private static CloseableHttpClient ourClient;
+	private static Server ourServer;
+	private static final FhirContext ourCtx = FhirContext.forR4();
+
+	private static final String TOKEN = "eyJhbGciOiJSUzI1NiIsInR5cCI6IkpXVCJ9.eyJzdWIiOiIxMjM0NTY3ODkwIiwibmFtZSI6IkpvaG4gRG9lIiwiYWRtaW4iOnRydWUsImlhdCI6MTUxNjIzOTAyMn0.NHVaYe26MbtOYhSKkoKYdFVomg4i8ZJd8_-RU8VNbftc4TSMb4bXP3l3YlNWACwyXPGffz5aXHc6lty1Y2t4SWRqGteragsVdZufDn5BlnJl9pdR_kdVFUsra2rWKEofkZeIC4yWytE58sMIihvo9H1ScmmVwBcQP6XETqYd0aSHp1gOa9RdUPDvoXQ5oqygTqVtxaDr6wUFKrKItgBMzWIdNZ6y7O9E0DhEPTbE9rfBo6KTFsHAZnMg4k68CDp2woYIaXbmYTWcvbzIuHO7_37GT79XdIwkm95QJ7hYC9RiwrV7mesbY4PAahERJawntho0my942XheVLmGwLMBkQ";
```

Comment:
Why is there a single hard-coded token?

Reply:
This is test token we have return below, as
 `DecodedJWT jwt = JWT.decode(token);` 
 method in CustomAuthorizationInterceptor requires a valid JWT to decode or we should mock this method as well.